### PR TITLE
[IMP] im_livechat: livechat send by email

### PR DESCRIPTION
**Pull request by mgh-odoo**
**Task**: https://www.odoo.com/web#id=2029660&model=project.task&menu_id=4720
**Pad**: https://pad.odoo.com/p/r.39aaa9fce3b87c2c184656047c77d4f0

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/im_livechat/static/src/js/im_livechat.js [hunk 1]
```diff
@@ -480,14 +482,43 @@ var Feedback = Widget.extend({
...
+        if ($email.val()) {
+            this._rpc({
+                route: '/im_livechat/email_livechat_transcript',
+                params: {
+                    res_id: this._livechat.getID(),
+                    email: $email.val(),
+                }
+            }).then(function () {
+                self.$('.o_livechat_email').html($('<strong />', { text: _t('Conversation Sent') }));
+            });
```
Comment (mge-odoo) [inline addons/im_livechat/static/src/js/im_livechat.js]: I would add a `guardedCatch` if something got wrong so the user has a feedback.

File: addons/im_livechat/static/src/js/im_livechat.js [hunk 2]
```diff
@@ -480,14 +482,43 @@ var Feedback = Widget.extend({
...
+            this._showThanksMessage();
+        }
+        this._sendFeedback({ close: false });
+    },
+    /**
+    * @private
+    */
+    _showThanksMessage: function () {
+        this.$('.o_livechat_rating_box').empty().append($('<div />', {
+            text: _t('Thankyou for your feedback'),
```
Comment (mge-odoo) [inline addons/im_livechat/static/src/js/im_livechat.js]:
```suggestion
            text: _t('Thank you for your feedback'),
```

File: addons/im_livechat/static/src/js/im_livechat.js [hunk 3]
```diff
@@ -480,14 +482,43 @@ var Feedback = Widget.extend({
...
+            class: 'text-muted'
+        }));
+    },
+    /**
+    * @private
+    * @param {MouseEvent} ev
+    */
+    _onEmailChat: function (ev) {
```
Comment (mge-odoo) [inline addons/im_livechat/static/src/js/im_livechat.js]: `ev` does not seem used

File: addons/im_livechat/static/src/xml/im_livechat.xml [hunk 4]
```diff
@@ -4,23 +4,34 @@
...
                 <textarea id="reason" placeholder="Explain your note"></textarea>
                 <div class="o_livechat_rating_reason_button">
                     <input type="button" class="o_rating_submit_button" value="Send" />
                 </div>
             </div>
+            <div class="o_livechat_email">
+                <span class="text-muted">Receive a copy of this conversation</span>
+                <div class="input-group">
+                    <input id="o_email" type="text" class="form-control"/>
+                    <button type="button" class="o_email_chat_button btn btn-primary rounded-0" value="email">
```
Comment (mge-odoo) [inline addons/im_livechat/static/src/xml/im_livechat.xml]: what is `value` attribute used for? seems useless

File: addons/im_livechat/static/src/js/im_livechat.js [hunk 5]
```diff
@@ -480,14 +482,55 @@ var Feedback = Widget.extend({
...
+                params: {
+                    uuid: this._livechat.getUUID(),
+                    email: $email.val(),
+                }
+            }).then(function () {
+                self.$('.o_livechat_email').html($('<strong />', { text: _t('Conversation Sent') }));
+            }).guardedCatch(function () {
+                self.$('.o_livechat_email').hide();
+                self.$('.o_livechat_email_error').show();
+                self.$('.o_livechat_email_error .alert-link').on('click', function () {
```
Comment (mge-odoo) [inline addons/im_livechat/static/src/js/im_livechat.js]: This event should be in the widget `events` dict, with a specific handler function.

File: addons/im_livechat/static/src/js/im_livechat.js [hunk 6]
```diff
@@ -480,14 +483,57 @@ var Feedback = Widget.extend({
...
-            shouldCloseChatWindow = true;
+            this._showThanksMessage();
         }
-        this._sendFeedback({ close: shouldCloseChatWindow });
+        this._sendFeedback({ close: false });
+    },
+    /**
+    * @private
+    */
+    _showThanksMessage: function () {
```
Comment (mge-odoo) [inline addons/im_livechat/static/src/js/im_livechat.js]: this method should no be placed in the Handlers section

**Comment by mge-odoo** [comment]
robodoo r+

**Comment by KangOl** [comment]
@robodoo retry

**Comment by d-fence** [comment]
robodoo retry

**Pull request closed without merging**